Fall_2020/Week_1/Reverse LinkedList.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+Problem: https://leetcode.com/problems/reverse-linked-list/
+
+Aproach: Reverse a singly linked list.
+
+Time complexity: O(n)
+Space complexity: O(n)
+
+*/
+#include <iostream>
+using namespace std;
+ struct ListNode {
+      int val;
+      ListNode *next;
+      ListNode() : val(0), next(nullptr) {}
+      ListNode(int x) : val(x), next(nullptr) {}
+      ListNode(int x, ListNode *next) : val(x), next(next) {}
+  };
+
+
+ ListNode* reverseList(ListNode* head) {
+        ListNode* prev = NULL;
+        while(head){
+            ListNode* next = head->next;
+            head->next = prev;
+            prev = head;
+            head = next;
+        }
+        cout << "Reverse Linked list: ";
+        cout << prev->val << "->";
+        cout << prev->next->val <<"->";
+        cout << prev->next->next->val << "->";
+        cout << prev->next->next->next->val << "->";
+        cout << prev->next->next->next->next->val << "->";
+        cout << prev->next->next->next->next->next;
```

Comment:
Please remove this and better create a function "printLinkedList", then call that function in main before and after the execution of reverseList().

Reply:
Maybe it can be similar to how the deleteList function mentioned on the other comment works.